clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -77,6 +84,13 @@
     private static final String SUBSCRIPTION_EXCEPTION_MESSAGE =
         "Subscription to topics, partitions and pattern are mutually exclusive";
 
+    private void confirmSubscriptionType(SubscriptionType type) {
+        if (this.subscriptionType.equals(SubscriptionType.NONE))
```

Comment:
Another one where I was debating what to choose. I simply picked `equals` because it is a built-in method. I can switch to using `==` if you think that reads better.